File: Part2_Memory_Management/Java/MemoryTest.java
class Box {
    int value;
    Box(int v) { value = v; }
}

public class MemoryTest
    static void square(Box b) {
        b.value = b.value * b.value; // object modified via reference
    }

    public static void main(String[] args) {
        Box num = new Box(5);
        square(num);
        System.out.println("Squared = " + num.value); // 25
    }
}

File: Part2_Memory_Management/Rust/MemoryTest.rs
fn square_in_place(x: &mut i32) { 
*x = *x * *x; 
} 
fn main() { 
let mut num = 5; 
// Pass mutable reference → borrow, no copy 
square_in_place(&mut num); 
println!("Squared = {}", num); // 25 
} // memory automatically freed here